config/env_sample.json

```diff
@@ -117,6 +117,53 @@
             }
         ]
     },
+    "LTI_CONFIG_TEMPLATE": {
```

Comment:
It doesn't look like this is something anyone would have to ever modify and everything is replaced in the code.

If it's just going to be a static json template we should probably just have it out in it's own file, part of the app separate from this json and config.

Reply:
I wholeheartedly agree!

I [asked on Slack for opinions](https://app.slack.com/client/T69CPPGNP/GNSJ3RMHN/thread/GNSJ3RMHN-1602174463.002200) about where the JSON template should go.

I wanted to support HJSON for the template file to allow comments in it.  But when I thought about the places in the project where the file could go, the `secrets` directory came to mind.  Since `env.json` already existed there, I thought it might as well go there.  It doesn't support HJSON (see: #1056), unfortunately, but it seems like a good place for it.

I also thought that there's a chance the format of the LTI config JSON could change in the future.  Either because of the way LTI is implemented or some of the Canvas-related values change.  So putting the LTI config template in the main app config makes it easier for others to update it as necessary.

Suggest a place for the template file to go and I'll put it there.

Reply:
I see, I haven't been looking at slack (outside of prod support) this week to focus on data studio. I see what you were talking about though. To me it feel like it should go in the dashboard folder as it's part of the app. Assets is where the current static file are stored (when collect static is run).

I'd probably put it in like `assets/config` or `assets/json` ? 

Then you should be able to just read it with `static("config/lti_config_template.json") `

Reply:
Yeah, I agree that the JSON template right now only covers the properties that we us in MyLA and we might want to add based on the needed. For Example `"default": "disabled",` this was added later as we did not want tool to be installed by default

Reply:
Having this as a configuration option should be a way to go `config/lti_config_template.json` so that in case of changes we can just change the configuration. But there might be certain options we might associate with the code

Reply:
Their might be certain LTI configuration options we might not have used might use in future having it configuration option is good and having it as separate file(`lti_config_template.json`) is even better so that it doesn't clutter the env.json as that file purpose is purely for properties that differ by env

Reply:
@pushyamig, I'm confused.

Which do you mean?

* [ ] Move LTI config template to `config/lti_config_template.json`
* [ ] Keep LTI config in `config/env_sample.json` (and therefore, the deployed `/secrets/env.json`)
* [ ] something else

Reply:
Thanks for clarifying!  I'll move it to `config/lti_config_template.json`.

Reply:
As I move this to another file, I had initially thought I would take the opportunity to add HJSON support to this.

However, as I think about it more, I have an opportunity to simplify this code.  Supporting HJSON for commenting and formatting probably won't be important for this file.  The comments would need to be parsed out before it could be sent to the requesting LMS anyway.  That would just add another step to the current process of replacing all the parameters in the string values and serializing everything back to JSON.

Instead, my thinking is to just use the new JSON file contents as one big string, apply formatting to it with the defined parameters, and return that to the LMS.  It wouldn't need the `recursive_format()` function at all.  That's a little disappointing, but it's probably worthwhile to reduce the complexity of the code.

Reply:
> Then you should be able to just read it with `static("config/lti_config_template.json") `

There were a few implementations of `static()` in Django, but none of them gave me the path to the file or its contents, as I was expecting.

In the documentation for Django's `staticfiles` app (https://docs.djangoproject.com/en/3.1/ref/contrib/staticfiles/#finders-module), I found that `finders.find()` would give me the full path to the file, so I use that like this:

```python
open(finders.find('config/lti_config_template.json')).read()
```